File: 07-architecture-for-team/android/app/src/main/java/com/chucknorrisviewer/AppModule.java
package com.chucknorrisviewer;


import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private final MainApplication app;

    AppModule(MainApplication app) {
        this.app = app;
    }

    @Singleton
    @Provides
    MainApplication provideMainApplication() {
        return app;
    }
}

File: 08-architecture-components/arch-mvvm/app/src/debug/java/jp/satorufujiwara/mvvm/di/AppModule.kt
package jp.satorufujiwara.mvvm.di

import dagger.Module
import dagger.Provides
import jp.satorufujiwara.mvvm.AppLifecycleCallbacks
import jp.satorufujiwara.mvvm.data.di.DataModule
import javax.inject.Singleton

@Module(includes = [DataModule::class])
class AppModule {

  @Singleton
  @Provides
  fun provideAppLifecycleCallbacks(): AppLifecycleCallbacks = DebugAppLifecycleCallbacks()

}